doc/services/portability/posix/option_groups/index.rst

```diff
@@ -102,123 +102,54 @@ This table lists service support status in Zephyr:
     pthread_getconcurrency(),yes
     pthread_setconcurrency(),yes
 
+.. _posix_option_group_c_lang_jump:
+
+POSIX_C_LANG_JUMP
+=================
+
+The ``POSIX_C_LANG_JUMP`` Option Group is included in the ISO C standard.
+
+.. note::
+   When using Newlib, Picolibc, or other C libraries conforming to the ISO C Standard, the
+   ``POSIX_C_LANG_JUMP`` Option Group is considered supported.
+
+.. csv-table:: POSIX_C_LANG_JUMP
+   :header: API, Supported
+   :widths: 50,10
+
+    setjmp(), yes
+    longjmp(), yes
+
+.. _posix_option_group_c_lang_math:
+
+POSIX_C_LANG_MATH
+=================
+
+The ``POSIX_C_LANG_MATH`` Option Group is included in the ISO C standard.
```

Comment:
POSIX has extensions to the ISO C specification, including numerous clarifications of math function behavior. Does Zephyr require conformance to POSIX specs or to the ISO C spec for these functions?

Reply:
Excellent question! The official answer is here
https://pubs.opengroup.org/onlinepubs/9699919799/help/codes.html#CX

TLDR - the ISO spec should be considered the correct one.

Reply:
This doesn't answer my question though -- for ISO C APIs where the POSIX specification places additional requirements on the implementation, can applications rely on the POSIX spec or must they only rely on the ISO C spec? That would require auditing all uses of ISO C APIs within any POSIX code brought into the Zephyr environment to ensure they do not rely on any POSIX extensions to the ISO C specification.

Given that POSIX specifications are not supposed to conflict with ISO C specifications, but only to refine them, it seems like Zephyr should require a POSIX conforming C library when the application requests POSIX functionality? Or at least make it clear to the application that they are operating outside known behavior when mixing non-POSIX implementations of C library functions with POSIX functionality.

Reply:
> This doesn't answer my question though -- for ISO C APIs where the POSIX specification places additional requirements on the implementation, can applications rely on the POSIX spec or must they only rely on the ISO C spec?

I think it means use the iso spec and consider the POSIX spec equivalent.

> That would require auditing all uses of ISO C APIs within any POSIX code brought into the Zephyr environment to ensure they do not rely on any POSIX extensions to the ISO C specification.

If you feel like doing that, I guess go for it?

Patches maybe welcome?

> Given that POSIX specifications are not supposed to conflict with ISO C specifications, but only to refine them, it seems like Zephyr should require a POSIX conforming C library when the application requests POSIX functionality?

Maybe. I'm ok with just using the ISO C spec for C standards though.

> Or at least make it clear to the application that they are operating outside known behavior when mixing non-POSIX implementations of C library functions with POSIX functionality.

Maybe. Again, I'm ok with just using the ISO C spec, and I think that's what the POSIX spec says to do too.

I can understand your concern though as a libc maintainer.

Reply:
@keith-packard - I'm not sure if there is any further clarification I can attempt to make here outside of what is already specified by IEEE.

Feel free to unresolve the conversation if you feel it's necessary. However, I think it's well outside the scope of this PR.